ios/Exponent/Versioned/Core/Api/EXScreenOrientation.h

```diff
@@ -1,14 +1,60 @@
 // Copyright 2015-present 650 Industries. All rights reserved.
 
-#import "EXScopedBridgeModule.h"
+#import <React/RCTBridgeModule.h>
+#import <React/RCTConvert.h>
+
+#import "EXScopedEventEmitter.h"
+
+typedef NS_ENUM (NSInteger, EXOrientation) {
+    PORTRAIT,
+    PORTRAIT_UP,
+    PORTRAIT_DOWN,
+    LANDSCAPE,
+    LANDSCAPE_LEFT,
+    LANDSCAPE_RIGHT,
+    UNKNOWN
+};
+
+typedef NS_ENUM(NSInteger, EXOrientationLock) {
+  DEFAULT_LOCK,
+  ALL_LOCK,
+  PORTRAIT_LOCK,
+  PORTRAIT_UP_LOCK,
+  PORTRAIT_DOWN_LOCK,
+  LANDSCAPE_LOCK,
+  LANDSCAPE_LEFT_LOCK,
+  LANDSCAPE_RIGHT_LOCK,
+  OTHER_LOCK,
+  ALL_BUT_UPSIDE_DOWN_LOCK // deprecated
+};
+
+@interface RCTConvert (OrientationLock)
+
++ (EXOrientationLock)EXOrientationLock:(id)json;
+
+@end
 
 @protocol EXScreenOrientationScopedModuleDelegate
 
 - (void)screenOrientationModule:(id)scopedOrientationModule
 didChangeSupportedInterfaceOrientations:(UIInterfaceOrientationMask)supportedInterfaceOrientations;
 
+- (UIInterfaceOrientationMask)supportedInterfaceOrientationsForVisibleApp;
+
+- (void)removeOrientationChangeListener:(NSString *) experienceId;
+
+- (void)addOrientationChangeListener:(NSString *) experienceId subscriberModule:(id) subscriberModule;
+
+- (UITraitCollection *) getTraitCollection;
+
 @end
 
-@interface EXScreenOrientation : EXScopedBridgeModule
+@interface EXScreenOrientation : EXScopedEventEmitter <RCTBridgeModule>
+
+- (void) handleScreenOrientationChange:(nullable UITraitCollection *) traitCollection;
 
++ (NSDictionary *)getStrToOrientationLockDict;
```

Comment:
+ use the "Native" and "JS" naming like the Android implementation does, which makes the underlying intent clearer. Also write out words like "String" in other places. In C89 abbreviating had a meaningful purpose (https://stackoverflow.com/questions/38035628/c-why-did-ansi-only-specify-six-characters-for-the-minimum-number-of-significa) but just about every language since then doesn't carry that baggage.

Reply:
@esamelson 
3/4 of the static methods i've moved to class properties
1/4 of the methods (getStrToOrientationLockDict) i left as a selector because i use it in RCT_ENUM_CONVERTER and it requires it to be exposed